src/app/pages/rpc-send-transfer/rpc-send-transfer.tsx

```diff
@@ -1,34 +1,245 @@
+import { useMemo, useState } from 'react';
```

Comment:
Maybe you can use a `.layout` file to help split this up between logic and presentation?

Reply:
There are components with many component props, can't they be in a layout component? 

In `getAddresses` the entire Approver component is in the layout, such that the root RPC compoent can be focused on business logic/event passing behaviours, and not layout and styles.